src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -1001,305 +993,17 @@ public unsafe bool TakeStartLine(ReadableBuffer buffer, out ReadCursor consumed,
             var overLength = false;
             if (buffer.Length >= ServerOptions.Limits.MaxRequestLineSize)
             {
-                bufferEnd = buffer.Move(start, ServerOptions.Limits.MaxRequestLineSize);
-
+                buffer = buffer.Slice(start, ServerOptions.Limits.MaxRequestLineSize);
```

Comment:
Have you considered handling limits in the parser implementation itself? Our current parser does this upfront, but what about a parser that keeps track of the count as it sees new bytes?